### PR TITLE
Pass X_train and y_train to graph_prediction_vs_actual_over_time

### LINKED ISSUE #2731: Update `graph_prediction_vs_actual_over_time` to use `predict_in_sample`
As of #2697, the user guide for time series does not use the `graph_prediction_vs_actual_over_time` function because it has not been updated to use the new `predict_in_sample` pipeline method.

image

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #2731 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -7,6 +7,9 @@ Release Notes
         * Set ``eval_metric`` to ``logloss`` for ``XGBoostClassifier`` :pr:`2741`
         * Added support for ``woodwork`` versions ``0.7.0`` and ``0.7.1`` :pr:`2743`
         * Changed ``explain_predictions`` functions to display original feature values :pr:`2759`
+        * Added ``X_train`` and ``y_train`` to ``graph_prediction_vs_actual_over_time`` and ``get_prediction_vs_actual_over_time_data`` :pr:`2762`
+        * Added ``forecast_horizon`` as a required parameter to time series pipelines and ``AutoMLSearch`` :pr:`2697`
```
Comment (freddyaboulton) [inline docs/source/release_notes.rst:11]: Accidentally put the release notes for 2697 into the previous release during a rebase lol

**Review by chukarsten — APPROVED** [review]
This LGTM!  Thank you!

**Review by jeremyliweishih — APPROVED** [review]
nice!

File: evalml/model_understanding/graphs.py [hunk 2]
```diff
@@ -1427,22 +1427,23 @@ def visualize_decision_tree(
     return source_obj
 
 
-def get_prediction_vs_actual_over_time_data(pipeline, X, y, dates):
+def get_prediction_vs_actual_over_time_data(pipeline, X, y, X_train, y_train, dates):
```
Comment (angela97lin) [inline evalml/model_understanding/graphs.py:1430]: Was this a bug? Or simply doing something different? If the latter, is there any functionality that we want to keep around?

**Review by angela97lin — APPROVED** [review]
Looks super clean! Just one question to feed my growing time-series knowledge 😂

File: evalml/model_understanding/graphs.py (hunk 2 above)
Comment (angela97lin) [reply evalml/model_understanding/graphs.py:1430]: Got it, cool stuff !

**Pull request merged**